tests/framework/matcher/getter.go

```diff
@@ -42,13 +42,16 @@ func ThisVMI(vmi *virtv1.VirtualMachineInstance) func() (*virtv1.VirtualMachineI
 
 // ThisVMIWith fetches the latest state of the VirtualMachineInstance based on namespace and name. If the object does not exist, nil is returned.
 func ThisVMIWith(namespace string, name string) func() (*virtv1.VirtualMachineInstance, error) {
```

Comment:
I am pretty sure this is also a problem for ThisVMWith and others

Reply:
true, I will add another commit for that fix :)

Reply:
We also have `ThisPodWith`, `ThisDVWith`, `ThisPVCWith`, `ThisMigrationWith` and `ThisDeploymentWith` - all with the same issue.

Reply:
Sorry there's actually more of them :( but totally understandable if it's too much for one PR
```bash
$ git grep -E 'func This.*[With]?\('
tests/framework/matcher/getter.go:func ThisPod(pod *v1.Pod) func() (*v1.Pod, error) {
tests/framework/matcher/getter.go:func ThisPodWith(namespace string, name string) func() (*v1.Pod, error) {
tests/framework/matcher/getter.go:func ThisVMI(vmi *virtv1.VirtualMachineInstance) func() (*virtv1.VirtualMachineInstance, error) {
tests/framework/matcher/getter.go:func ThisVMIWith(namespace string, name string) func() (*virtv1.VirtualMachineInstance, error) {
tests/framework/matcher/getter.go:func ThisVM(vm *virtv1.VirtualMachine) func() (*virtv1.VirtualMachine, error) {
tests/framework/matcher/getter.go:func ThisVMWith(namespace string, name string) func() (*virtv1.VirtualMachine, error) {
tests/framework/matcher/getter.go:func ThisDV(dv *v1beta1.DataVolume) func() (*v1beta1.DataVolume, error) {
tests/framework/matcher/getter.go:func ThisDVWith(namespace string, name string) func() (*v1beta1.DataVolume, error) {
tests/framework/matcher/getter.go:func ThisPVC(pvc *v1.PersistentVolumeClaim) func() (*v1.PersistentVolumeClaim, error) {
tests/framework/matcher/getter.go:func ThisPVCWith(namespace string, name string) func() (*v1.PersistentVolumeClaim, error) {
tests/framework/matcher/getter.go:func ThisMigration(migration *virtv1.VirtualMachineInstanceMigration) func() (*virtv1.VirtualMachineInstanceMigration, error) {
tests/framework/matcher/getter.go:func ThisMigrationWith(namespace string, name string) func() (*virtv1.VirtualMachineInstanceMigration, error) {
tests/framework/matcher/getter.go:func ThisDeploymentWith(namespace string, name string) func() (*k8sv1.Deployment, error) {
```

Reply:
done